File: NodeJS/Embed for your customers/AppOwnsData/models/embedConfig.js
// ----------------------------------------------------------------------------
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.
// ----------------------------------------------------------------------------

// Properties for embedding the report 
class EmbedConfig {
    constructor(type, reportsDetail, embedToken) {
        this.type = type;
        this.reportsDetail = reportsDetail;
        this.embedToken = embedToken;
    }
}

module.exports = EmbedConfig;

File: Python/Embed for your customers/AppOwnsData/models/embedconfig.py
# Copyright (c) Microsoft Corporation.
# Licensed under the MIT license

class EmbedConfig:

    # Camel casing is used for the member variables as they are going to be serialized and camel case is standard for JSON keys

    tokenId = None
    accessToken = None
    tokenExpiry = None
    reportConfig = None

    def __init__(self, token_id, access_token, token_expiry, report_config):
        self.tokenId = token_id
        self.accessToken = access_token
        self.tokenExpiry = token_expiry
        self.reportConfig = report_config